benches/nexmark/generator/auctions.rs

```diff
@@ -11,6 +11,55 @@ use std::{
 };
 
 impl<R: Rng> NexmarkGenerator<R> {
+    /// Return the last valid auction id (ignoring FIRST_AUCTION_ID). Will be
+    /// the current auction id if due to generate an auction.
+    fn last_base0_auction_id(&self, event_id: u64) -> u64 {
+        let mut epoch = event_id / self.config.nexmark_config.total_proportion() as u64;
+        let mut offset = event_id % self.config.nexmark_config.total_proportion() as u64;
+
+        if offset < self.config.nexmark_config.person_proportion as u64 {
+            // About to generate a person.
+            // Go back to the last auction in the last epoch.
+            epoch = match epoch.checked_sub(1) {
+                Some(e) => e,
+                None => return 0,
+            };
+            offset = self.config.nexmark_config.auction_proportion as u64 - 1;
+        } else if offset
+            >= (self.config.nexmark_config.person_proportion
+                + self.config.nexmark_config.auction_proportion) as u64
+        {
+            // About to generate a bid.
+            // Go back to the last auction generated in this epoch.
+            offset = self.config.nexmark_config.auction_proportion as u64 - 1;
+        } else {
+            // About to generate an auction.
+            offset -= self.config.nexmark_config.person_proportion as u64;
+        }
+        epoch * self.config.nexmark_config.auction_proportion as u64 + offset
```

Comment:
Maybe do these casts once and store them in variables

Reply:
Yeah, I was wondering whether I should just use u64 for them. It seemed odd to choose u64 for something I know is small in the configuration, but given that they're consistently used in expressions for the (long) ids, perhaps it'd be better to just switch them to u64 in the config?

Reply:
Actually, it makes it much more readable to create variables for each (rather than reaching through `self.config.nexmark_config.` each time) anyway, so I'll just do as you suggested :)